src/sage/rings/polynomial/laurent_polynomial_mpair.pyx

```diff
@@ -1168,7 +1169,17 @@ cdef class LaurentPolynomial_mpair(LaurentPolynomial):
             1
             sage: f.degree(z)
             0
+
+        The zero polynomial is defined to have degree `-\infty`::
+
+            sage: R.<x> = LaurentPolynomialRing(ZZ)
+            sage: R.zero().degree()
+            -Infinity
         """
+        # The zero polynomial is defined to have degree -Infinity
+        if self.is_zero():
+            return minus_infinity
+
         if not x:
```

Comment:
This should actually be `x is None`, and documented.

Reply:
I'm sorry I don't understand. `x` here is an optional argument which looks for the degree of the generator `x` in the polynomial in `f(x, y, z, ...)`.

Are you saying we should document that both `R.zero().degree() = R.zero().degree(x) = -Infinity`>

Reply:
Thank you! Actually, I didn't realise that it would return the degree of the highest degree term, which seems extremely odd to me.  I would have thought that it would return total degree in this case.  Did you check whether this is used anywhere?

Reply:
Maybe I am misunderstanding and it returns the total degree and my docstring example is confusing?

What is the total degree of a Laurent Polynomial Ring? Sorry for my naivety-- this isn't an area I work in much.

Reply:
I think my understanding of the current behaviour is correct, for both the univariate and multivariate cases the degree is the degree of the term with the highest degree.

```py
sage: R.<x> = LaurentPolynomialRing(ZZ)
sage: (x^3 + x).degree()
3
sage: (x^-3 + x).degree()
1
sage: 
sage: R.<x, y> = LaurentPolynomialRing(ZZ)
sage: (x + y^3).degree()
3
sage: (x*y^6 + y^3).degree()
7
```

Following this logic, in PR https://github.com/sagemath/sage/pull/37490 I added valuation for the multivariate case and this behaves similarly to degree:

```py
    def valuation(self, x=None):
         """
         Return the valuation of ``x`` in ``self``.
         If ``x`` is ``None``, return the minimal valuation of ``self``.
         EXAMPLES::
             sage: R.<x,y> = LaurentPolynomialRing(ZZ)
             sage: f = 2*x^2*y^-3 - 13*x^-1*y^-3 + 2*x^2*y^-5 - 2*x^-3*y^2
             sage: f.valuation()
             -4
             sage: f.valuation(x)
             -3
             sage: f.valuation(y)
             -5
             sage: R.zero().valuation()
             +Infinity
```

and matches the behaviour of the univariate case

```py
sage: R.<x> = LaurentPolynomialRing(ZZ)
sage: (x^-3 + x).valuation()
-3
```

Reply:
Oh! but a comment from @nbruin in this PR says that valuation is a bad term for this, so maybe I will change this anyway!

I'm not sure I'm the one to be dealing with all these PR!

Reply:
I did not doubt that your interpretation of the code was correct, I was just surprised.  I must admit, surprised negatively, because this conflicts with the definition used in `LazyPowerSeriesRing`.

I think we first have to sort out the mathematics involved.

Reply:
I actually wonder if there's a bug here as the underlying call is poly._total_degree() which is using singular. 

This area of mathematics is not my speciality so there's many things I may misunderstand but I'm happy to try and help. 

Maybe worth first digging into what .degree() is trying to compute (maybe the multivariate total_degree() is weirdly behaved?